### PR TITLE
Overlay prebase

### PR DESCRIPTION
**Pull request by mdurrant-b3**

### CODE REVIEW

File: acos_client/v30/overlay/vtep.py [hunk 1]
```diff
@@ -95,18 +96,21 @@ def _add_source_vni(self, vtep_id, ip_address, segment, partition=None, gateway=
...
-    def _build_ip_payload_and_url(self, vtep_id, target, ip_address, encap_type):
+    def _build_ip_payload_and_url(self, vtep_id, target, ip_address, encap_type, vni, lif_id=None):
         addr_type = "{0}-ip-address".format(target)
 
         payload = {
             addr_type: {
                 "ip-address": ip_address,
             }
         }
+        import pdb; pdb.set_trace()
```
Comment (jak3kaj) [inline acos_client/v30/overlay/vtep.py]: Probably don't want to leave this in.

File: acos_client/v30/overlay/vtep.py [hunk 2]
```diff
@@ -54,17 +54,18 @@ def create(self, vtep_id, source_ip=None, source_vnis=[],
...
                 "vtep": {
                     "id": vtep_id,
-                    "encap": encap_type,
                 }
             }
             existing = self._post(self.url_prefix, payload, **kwargs)
-
+        # iterate later.
+        dest_ip = dest_ips[0]
+        vni = source_vnis=[0]
```
Comment (jak3kaj) [inline acos_client/v30/overlay/vtep.py]: You mean `vni = source_vnis[0]`, right?

File: acos_client/v30/base.py [hunk 3]
```diff
@@ -59,6 +60,8 @@ def _request(self, method, action, params, retry_count=0, **kwargs):
                 return self._request(method, action, params, retry_count+1,
                                      **kwargs)
             raise e
+#        except json.JSONDecodeError as jsonex:
```
Comment (jak3kaj) [inline acos_client/v30/base.py]: Was this added for debugging and should be removed, or is it something you want to implement later?

File: acos_client/v30/overlay/vtep.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            # Create it.
+            payload = {
+                "vtep": {
+                    "id": vtep_id,
+                }
+            }
+            existing = self._post(self.url_prefix, payload, **kwargs)
+        # iterate later.
+        dest_ip = dest_ips[0]
+        vni = source_vnis=[0]
```
Comment (jak3kaj) [inline acos_client/v30/overlay/vtep.py]: should be `source_vnis[0]` right?

File: acos_client/v30/base.py (hunk 3 above)
Comment (mdurrant-b3) [reply acos_client/v30/base.py]: Should be removed. I'll followup with a patch to delete this comment.

**Pull request merged**